cmd/root.go

```diff
@@ -249,11 +280,52 @@ func parseConfig(cmd *cobra.Command, conf *proxy.Config, args []string) error {
 	return nil
 }
 
+// configureTelemetry configures Cloud Trace and Cloud Monitoring based on user
+// selection of each feature. If neither have been enabled, this function is a
+// no-op.
+func configureTelemetry(cmd *Command) (func(), error) {
+	if !cmd.telemetryTracing && !cmd.telemetryMetrics {
+		return func() {}, nil
+	}
```

Comment:
nit: Seems like maybe you could do this check outside and just avoid calling the function if true